### PR TITLE
refactor(message): refactor to cssinjs

### PR DESCRIPTION
**Pull request by TrickyPi**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Refactor message to cssinjs  |
| 🇨🇳 Chinese |   重构message为cssinjs        |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/message/useMessage.tsx [hunk 1]
```diff
@@ -95,7 +98,7 @@ const Holder = React.forwardRef<HolderRef, HolderProps>((props, ref) => {
     prefixCls,
   }));
 
-  return holder;
+  return wrapSSR(holder);
```
Comment (TrickyPi) [inline components/message/useMessage.tsx:101]: 这个是不是可以不包一层

File: components/_util/theme/interface.ts [hunk 2]
```diff
@@ -370,4 +372,6 @@ export interface AliasToken extends Omit<DerivativeToken, OmitDerivativeKey> {
 
   colorPopupBg: string;
   colorBorderSecondary: string;
+
+  messageNoticeContentPadding: string;
```
Comment (TrickyPi) [inline components/_util/theme/interface.ts:376]: 这个token发现在compact和default主题下不一样，可以放在这里不
原先的[less变量](https://github.com/ant-design/ant-design/blob/master/components/message/style/index.less#L23)
default下的[值](https://github.com/ant-design/ant-design/blob/master/components/style/themes/default.less#L943)
compact下的[值](https://github.com/ant-design/ant-design/blob/master/components/style/themes/compact.less#L196)

File: components/message/style/index.tsx [hunk 3]
```diff
@@ -1,2 +1,145 @@
...
+  } = token;
+  return {
+    [componentCls]: {
+      ...resetComponent(token),
+      position: 'fixed',
+      top: '8px',
+      insetInlineStart: 0, // affected by ltr or rtl
+      width: '100%',
+      pointerEvents: 'none',
+      zIndex: zIndexMessage,
```
Comment (TrickyPi) [inline components/message/style/index.tsx:68]: `@A`的位置

File: components/message/style/index.tsx [hunk 4]
```diff
@@ -1,2 +1,145 @@
...
+import { genComponentStyleHook, resetComponent, mergeToken } from '../../_util/theme';
+
+/** Component only token. Which will handle additional calculation of alias token */
+export interface ComponentToken {
+  // Component token here
+}
+
+interface MessageToken extends FullToken<'Message'> {
+  // Custom token here
+  zIndexMessage: number;
```
Comment (TrickyPi) [inline components/message/style/index.tsx:14]: 原先的[less变量](https://github.com/ant-design/ant-design/blob/master/components/message/style/index.less#L12)，这个less变量在default.less和variable.less里面定义的，但是值是一样的，我们是不是可以直接在`@A`位置，写死为1010.

File: components/message/style/index.tsx [hunk 5]
```diff
@@ -1,2 +1,145 @@
...
+}
+
+const messageMoveIn = new Keyframes('MessageMoveIn', {
+  '0%': {
+    padding: 0,
```
[the remark this answers is not in the document]
Comment (TrickyPi) [reply components/message/style/index.tsx:19]: 为啥只有数字类需要打上FIXME...

File: components/message/useMessage.tsx [hunk 6]
```diff
@@ -56,14 +57,16 @@ const Holder = React.forwardRef<HolderRef, HolderProps>((props, ref) => {
...
+
   // =============================== Style ===============================
   const getStyle = () => ({
     left: '50%',
     transform: 'translateX(-50%)',
     top: top ?? DEFAULT_OFFSET,
   });
 
-  const getClassName = () => (rtl ? `${prefixCls}-rtl` : '');
+  const getClassName = () => `${rtl ? `${prefixCls}-rtl` : ''} ${hashId}`;
```
[the remark this answers is not in the document]
Comment (TrickyPi) [reply components/message/useMessage.tsx]: done

File: components/message/style/index.tsx [hunk 7]
```diff
@@ -1,2 +1,145 @@
...
+      },
+      [`
+        ${componentCls}-move-up-appear${componentCls}-move-up-appear-active,
+        ${componentCls}-move-up-enter${componentCls}-move-up-enter-active
+      `]: {
+        animationPlayState: 'running',
+      },
+      [`${componentCls}-move-up-leave`]: {
+        animationName: messageMoveOut,
+        animationDuration: '0.3s',
```
[the remark this answers is not in the document]
Comment (TrickyPi) [reply components/message/style/index.tsx]: done

File: components/message/style/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (TrickyPi) [reply components/message/style/index.tsx:19]: done

File: components/_util/theme/interface.ts (hunk 2 above)
Comment (MadCcc) [reply components/_util/theme/interface.ts:385]: 不用管 compact，以 default 为准。另外这里不要加组件的 token，都写在 custom token 里

File: components/message/style/index.tsx (hunk 4 above)
Comment (MadCcc) [reply components/message/style/index.tsx:14]: 用 `token.zIndexBase + 10`

File: components/message/style/index.tsx [hunk 8]
```diff
@@ -1,2 +1,155 @@
...
+    radiusBase,
+    // Custom token
+    zIndexMessage,
+  } = token;
+  return {
+    [componentCls]: {
+      ...resetComponent(token),
+      position: 'fixed',
+      // FIXME: hard code in v4
+      top: '8px',
```
Comment (MadCcc) [inline components/message/style/index.tsx:69]:
```suggestion
      top: token.marginXS,
```

File: components/message/style/index.tsx [hunk 9]
```diff
@@ -1,2 +1,155 @@
...
+        position: 'relative',
+        // FIXME: hard code in v4
+        top: '1px',
+        // FIXME: hard code in v4
+        marginInlineEnd: '8px', // affected by ltr or rtl
+        fontSize: fontSizeLG,
+      },
+      [`${componentCls}-notice`]: {
+        // FIXME: hard code in v4
+        padding: '8px',
```
Comment (MadCcc) [inline components/message/style/index.tsx:86]: 8px 都可以用 paddingXS 或者 marginXS 替代

Comment (TrickyPi) [reply components/message/style/index.tsx:86]: marginInlineEnd是不是也可以用marginXS

File: components/_util/theme/interface.ts (hunk 2 above)
Comment (TrickyPi) [reply components/_util/theme/interface.ts]: done

File: components/message/style/index.tsx (hunk 9 above)
Comment (MadCcc) [reply components/message/style/index.tsx]: 对的，能用 token 的都用 token

File: components/message/style/index.tsx [hunk 10]
```diff
@@ -1,2 +1,155 @@
...
+
+interface MessageToken extends FullToken<'Message'> {
+  // Custom token here
+  zIndexMessage: number;
+}
+
+// FIXME: hard code in v4
+const messageMoveIn = new Keyframes('MessageMoveIn', {
+  '0%': {
+    padding: 0,
```
[the remark this answers is not in the document]
Comment (TrickyPi) [reply components/message/style/index.tsx]: done

File: components/message/style/index.tsx [hunk 11]
```diff
@@ -1,2 +1,155 @@
...
+    padding: '8px',
+    transform: 'translateY(0)',
+    opacity: 1,
+  },
+});
+
+// FIXME: hard code in v4
+const messageMoveOut = new Keyframes('MessageMoveOut', {
+  '0%': {
+    maxHeight: '150px',
```
[the remark this answers is not in the document]
Comment (TrickyPi) [reply components/message/style/index.tsx]: done

File: components/message/style/index.tsx (hunk 9 above)
Comment (TrickyPi) [reply components/message/style/index.tsx]: 好的

File: components/message/style/index.tsx (hunk 8 above)
Comment (TrickyPi) [reply components/message/style/index.tsx:70]: 语义上，感觉怪怪的

Comment (MadCcc) [reply components/message/style/index.tsx:70]: 代表间距的值，语义上也算贴近吧

Comment (TrickyPi) [reply components/message/style/index.tsx:70]: 嗯嗯，好的吧

**Comment by MadCcc** [comment]
conflict

**Comment by TrickyPi** [comment]
嗯嗯，解决了

**Pull request merged**